packages/flutter_availability/lib/src/ui/screens/template_day_modification.dart

```diff
@@ -28,25 +28,14 @@ class DayTemplateModificationScreen extends StatefulWidget {
 
 class _DayTemplateModificationScreenState
     extends State<DayTemplateModificationScreen> {
-  late int? _selectedColor;
-  late AvailabilityTemplateModel _template;
+  late DayTemplateViewModel _viewModel;
 
   @override
   void initState() {
     super.initState();
-    _selectedColor = widget.template?.color;
-    _template = widget.template ??
-        AvailabilityTemplateModel(
-          userId: "1",
-          name: "",
-          color: 0,
-          templateType: AvailabilityTemplateType.day,
-          templateData: DayTemplateData(
-            startTime: DateTime.now(),
-            endTime: DateTime.now(),
-            breaks: [],
-          ),
-        );
+    _viewModel = widget.template != null
```

Comment:
We seem to do this a lot, I feel like this can be done better.
```
_value = someValue != null ? TypeDependingOnValue.of(someValue) : null;
```

These always felt hard to read to me